cdap-ui/app/directives/timeline/timeline-ctrl.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+function TimelineController ($scope, LogViewerStore) {
+  console.log('From TimelineController: ' + LogViewerStore.getState().startTime);
+
+  this.testThis = 'hello';
+
+  LogViewerStore.subscribe(() => {
+    this.testTime = LogViewerStore.getState().startTime;
+  });
+
+  // LogViewerStore.subscribe(() => {
+  //   this.testScroll = LogViewerStore.getState();
+  // });
+
+  this.updateTheSlider = function() {
+    LogViewerStore.dispatch({
```

Comment:
I don't think this function is used. In the link function, when we change the slider we need to call this function to update the store and the subscribers. This is again part of stitching things together, the view - store & the controller.